R/add-subvariable.R

```diff
@@ -67,3 +75,128 @@ addSubvarDef <- function(var, subvar) {
     }
     return(as.character(out))
 }
+
+
+addSubvariableDerived <- function(variable, subvariable) {
+    if (is.VarDef(subvariable) | is.variable(subvariable)) subvariable <- list(subvariable)
+    if (is.catalog(subvariable) | is.dataset(subvariable)) subvariable <- lapply(
+        seq_along(subvariable),
+        function(var_num) subvariable[[var_num]]
+    )
+    # bypass `derivation(variable)` because `select` zcl function has ids
+    # not urls and so absolutifyURL mangles url
+    # TODO: use `derivation()` when select has relative urls (pivotal ticket: ???)
+    old_deriv <- CrunchExpr(expression = entity(variable)@body$derivation)
+
+    if (isSelectDerivation(old_deriv)) {
+        new_deriv <- addToSelectDerivation(old_deriv, subvariable)
+    } else if (isSelectCatDerivation(old_deriv)) {
+        new_deriv <- addToSelectCatDerivation(old_deriv, subvariable, categories(variable))
+    } else {
+        halt("Could not add subvariable because did not recognize variable derivation structure")
+    }
+
+    derivation(variable) <- new_deriv
+    # We don't get metadata from original variable like we would if we were creating
+    # subvariable during original derivation...
+    # TODO: It would be nice to update name in the same POST as updating derivation
+    # to ensure consistency, but I don't see how yet
+    # TODO: Also, the alias is really ugly for newly created subvariables
```

Comment:
Is there a way to get the server's naming logic? I tried just adding the alias explicitly, hoping that the server would dedup for me, but this resulted in a failure.